File: 240_search_a_2D_matrix_2/searchMatrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m = matrix.length;
        int n = matrix[0].length;

        if (matrix == null || m == 0 || n == 0) {
            return false;
        }

        int c = 0;
        int r = n - 1;

        while (c < m && r >= 0) {
            if (matrix[c][r] == target) {
                return true;
            }

            if (matrix[c][r] > target) {
                r--;
            } else {
                c++;
            }
        }

        return false;
    }
}

File: 240_search_a_2D_matrix_2/searchMatrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        n = len(matrix)
        m = len(matrix[0])

        if matrix == None or n == 0 or m == 0:
            return False

        col = 0
        row = m - 1

        while col < n and row >= 0:
            if matrix[col][row] == target:
                return True
            if matrix[col][row] > target:
                row -= 1
            else:
                col += 1

        return False